auth/src/main/java/com/firebase/ui/auth/util/ui/FlowUtils.java

```diff
@@ -0,0 +1,43 @@
+package com.firebase.ui.auth.util.ui;
+
+import android.app.Activity;
+import android.app.PendingIntent;
+import android.content.IntentSender;
+import android.support.annotation.NonNull;
+
+import com.firebase.ui.auth.IdpResponse;
+import com.firebase.ui.auth.data.model.IntentRequiredException;
+import com.firebase.ui.auth.data.model.PendingIntentRequiredException;
+import com.firebase.ui.auth.data.model.StatefulException;
+import com.firebase.ui.auth.ui.HelperActivityBase;
+
+public final class FlowUtils {
+    private FlowUtils() {
+        throw new AssertionError("No instance for you!");
+    }
+
+    public static boolean handleError(@NonNull HelperActivityBase activity, @NonNull Exception e) {
+        if (!(e instanceof StatefulException)) { return false; }
```

Comment:
Not sure that `StatefulException` is really a useful / meaningful base class (as stated in a comment above).  Why not just return `true` inside each of the `if` blocks and then return `false` at the end otherwise?

Reply:
Totally agree, been there, done that. 👍😆